### PR TITLE
Versioning unification, dependencies update, logging optimization

### PR DESCRIPTION
**Pull request by rmalmain**
- versioning unification -> use the x.y.z everywhere (https://doc.rust-lang.org/cargo/reference/resolver.html#recommendations).
- do not compile low-level logs (< info level) by default in example fuzzers
- update dependencies to latest versions

it's wip, i'll wait feedback to continue. any opinion on this? (cc @domenukk @tokatoka)

### CODE REVIEW

**Comment by tokatoka** [comment]
maybe you shouldn't do this before we merge your fuzzer dir structure stuff

**Comment by rmalmain** [comment]
yep it's also why i didn't continue for now.
the merge should be smooth though, since i only moved folders around.

File: libafl/Cargo.toml [hunk 1]
```diff
@@ -247,42 +249,42 @@ tokio = { version = "1.38", optional = true, features = [
...
-document-features = { version = "0.2", optional = true }
+document-features = { version = "0.2.10", optional = true }
 # Optional
-clap = { version = "4.5", optional = true }
+clap = { version = "4.5.18", optional = true }
 
 [target.'cfg(unix)'.dependencies]
-libc = "0.2"                                 # For (*nix) libc
-z3 = { version = "0.12.0", optional = true } # for concolic mutation
+libc = "0.2.159"                             # For (*nix) libc
```
Comment (domenukk) [inline libafl/Cargo.toml:283]: If we specify that stuff so closely, we should start using workspace dependencies instead so we don't have multiple crates with different dependency versions

File: libafl_frida/src/utils.rs [hunk 2]
```diff
@@ -228,38 +228,33 @@ pub(crate) fn frida_to_cs(
 /// Get the `base`, `idx`, `scale`, `disp` for each operand
 pub fn operand_details(operand: &Operand) -> Option<(X86Register, X86Register, u8, i32)> {
     match operand {
-        Operand::RegDeref(base) => {
```
Comment (domenukk) [inline libafl_frida/src/utils.rs:231]: @s1341 can you see if that's all right?

**Comment by rmalmain** [comment]
this one is ready.
`frida-gum` version `0.13.7` [seems to have a problem with clippy](https://github.com/AFLplusplus/LibAFL/actions/runs/11204078318/job/31142122206), when `DOCS_RS=1` and `auto-download` feature is disabled (cc @s1341, can you have a look?).

**Comment by s1341** [comment]
I think I know the issue... will try to fix it today.

**Comment by rmalmain** [comment]
> I think I know the issue... will try to fix it today.

thank you! btw, i made minor modifications to frida to make it compatible with the latest version of `yaxpeax-x86` (cf @domenukk comment above), does it look good to you?

File: libafl_frida/src/utils.rs (hunk 2 above)
Comment (s1341) [reply libafl_frida/src/utils.rs:231]: I don't have time to test at the moment. We'll see if it breaks things :| yolo, right?

Comment (rmalmain) [reply libafl_frida/src/utils.rs:231]: yoloo

File: libafl_bolts/src/minibsod.rs [hunk 3]
```diff
@@ -1199,7 +1206,7 @@ mod tests {
         let t = std::thread::spawn(move || {
             let cur = unsafe { GetCurrentThread() };
             let proc = unsafe { GetCurrentProcess() };
-            let mut out = Default::default();
+            let mut out: HANDLE = HANDLE::default();
```
Comment (domenukk) [inline libafl_bolts/src/minibsod.rs:1209]: We don't need to specify this dude twice right?

File: scripts/parallellize_cargo_check.py [hunk 4]
```diff
@@ -38,8 +38,9 @@
     print(os.environ)
     if "libafl_frida" in task:
         # DOCS_RS is needed for libafl_frida to build without auto-download feature
-        cargo_check = subprocess.check_output(
-            task, shell=True, text=True, env=dict(os.environ, DOCS_RS="1")
-        )
+        # cargo_check = subprocess.check_output(
```
Comment (domenukk) [inline scripts/parallellize_cargo_check.py:41]: What happened to this check?

**Review by domenukk — APPROVED** [review]
Awesome, let's merge

File: scripts/parallellize_cargo_check.py (hunk 4 above)
Comment (rmalmain) [reply scripts/parallellize_cargo_check.py:41]: https://github.com/AFLplusplus/LibAFL/actions/runs/11204078318/job/31142122206

we can keep it this way for now, once @s1341 finished to fix the problem we can open another pr to re-enable clippy for frida.

File: libafl_nyx/build_nyx_support.sh [hunk 5]
```diff
@@ -40,7 +43,7 @@ test -e QEMU-Nyx/.git || { echo "[-] QEMU-Nyx not checked out, please install gi
 echo "[*] Checking QEMU-Nyx ..."
 if [ ! -f "QEMU-Nyx/x86_64-softmmu/qemu-system-x86_64" ]; then
     cd QEMU-Nyx/ || return
-    cp ../Makefile.libxdc ./libxdc/Makefile || exit 1
+    sed -i "s,git submodule update libxdc$,git submodule update libxdc \&\& cp ../Makefile.libxdc ./libxdc/Makefile || exit 1," compile_qemu_nyx.sh
```
[the remark this answers is not in the document]
Comment (rmalmain) [reply libafl_nyx/build_nyx_support.sh:46]: sure

File: scripts/parallellize_cargo_check.py (hunk 4 above)
Comment (domenukk) [reply scripts/parallellize_cargo_check.py:41]: I fear we'll just forget it. Can't we keep it broken?

Comment (rmalmain) [reply scripts/parallellize_cargo_check.py:41]: ok i don't mind

**Comment by domenukk** [comment]
image
That's the spirit :D

**Pull request merged**